src/main/java/carpet/script/value/Value.java

```diff
@@ -79,6 +81,14 @@ public String getPrettyString()
 
     public Value add(Value o) {
         String lstr = this.getString();
+        if (o instanceof FormattedTextValue)
+        {
+            if (lstr == null)
+                return ((FormattedTextValue) o).clone();
```

Comment:
since this piece of code is same/similar than the .add() in FormattedText, why don't you have static combine(Formatted 1, Formatted 2) in FormattedTextValue and use that here and in .add()